src/Storages/PostgreSQL/MaterializedPostgreSQLConsumer.cpp

```diff
@@ -515,13 +525,14 @@ void MaterializedPostgreSQLConsumer::processReplicationMessage(const char * repl
 
 void MaterializedPostgreSQLConsumer::syncTables()
 {
-    try
+    for (const auto & table_name : tables_to_sync)
     {
-        for (const auto & table_name : tables_to_sync)
-        {
-            auto & storage_data = storages.find(table_name)->second;
-            Block result_rows = storage_data.buffer.description.sample_block.cloneWithColumns(std::move(storage_data.buffer.columns));
+        auto & storage_data = storages.find(table_name)->second;
+        Block result_rows = storage_data.buffer.description.sample_block.cloneWithColumns(std::move(storage_data.buffer.columns));
+        storage_data.buffer.columns = storage_data.buffer.description.sample_block.cloneEmptyColumns();
```

Comment:
the fix is in moving this line right after doing `cloneWithColumns(std::move(storage_data.buffer.columns))`, because previously it was done after data sync, and if exception occurred -- this would not be done and `storage_data.buffer.columns` will be left uninitialized.